user_tools/src/spark_rapids_tools/utils/util.py

```diff
@@ -160,3 +166,62 @@ def init_environment(short_name: str):
     print(Utils.gen_report_sec_header('Application Logs'))
     print(f'Location: {log_file}')
     print('In case of any errors, please share the log file with the Spark RAPIDS team.\n')
+
+
+class Utilities:
+    """Utility class used to enclose common helpers and utilities."""
+
+    @classmethod
+    def get_latest_mvn_jar_from_metadata(cls, url_base: str,
+                                         loaded_version: str = None) -> str:
+        """
+        Given the defined version in the python tools build, we want to be able to get the highest
+        version number of the jar available for download from the mvn repo.
+        The returned version is guaranteed to be LEQ to the defined version. For example, it is not
+        allowed to use jar version higher than the python tool itself.
+
+        The implementation relies on parsing the "$MVN_REPO/maven-metadata.xml" which guarantees
+        that any delays in updating the directory list won't block the python module
+        from pulling the latest jar.
+
+        :param url_base: the base url from which the jar file is downloaded. It can be mvn repo.
+        :param loaded_version: the version from the python tools in string format
+        :return: the string value of the jar that should be downloaded.
+        """
+
+        if loaded_version is None:
+            loaded_version = cls.get_base_release()
+        context = ssl.create_default_context(cafile=certifi.where())
+        defined_version = Version(loaded_version)
+        jar_version = Version(loaded_version)
+        xml_path = f'{url_base}/maven-metadata.xml'
+        with urllib.request.urlopen(xml_path, context=context) as resp:
+            xml_content = resp.read()
+            xml_root = elem_tree.fromstring(xml_content)
+            for version_elem in xml_root.iter('version'):
+                curr_version = Version(version_elem.text)
+                if curr_version <= defined_version:
```

Comment:
nit: can we use `cur_version <= jar_version` here, so that we can remove the variable `defined_version`.